### PR TITLE
New sparkl cd command

### PR DESCRIPTION
**Pull request by jacobythwaites**
Implementation of `cd` command, adjusting all existing commands that specify folders to use the current folder as base.

### CODE REVIEW

File: sparkl_cli/cmd_cd.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    put_connection)
+
+
+def parse_args(subparser):
+    """
+    Module-specific subcommand arguments.
+    """
+    subparser.add_argument(
+        "folder",
+        nargs="?",
```
Comment (miklosduma) [inline sparkl_cli/cmd_cd.py]: I would add "default=/". You would expect an empty cd to return to your home folder.

File: sparkl_cli/cmd_cd.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        help="folder to move into relative to current folder")
+
+
+def command(args):
+    """
+    Shows the current folder if no argument supplied, otherwise
+    changes to the specified folder.
+    """
+    current = get_current_folder(args)
+    if args.folder:
```
Comment (miklosduma) [inline sparkl_cli/cmd_cd.py]: If you add a default path for the parser (e.g. root folder) you do not have to write conditionals against it

File: sparkl_cli/test/test_cd.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            "Scratch/TestPublicRest",
+            alias="pytest_session")
+
+        sparkl(
+            "logout",
+            alias="pytest_session")
+
+        sparkl(
+            "close",
+            all=True,
```
Comment (miklosduma) [inline sparkl_cli/test/test_cd.py]: IF you specify an alias, why close all too?

File: sparkl_cli/cmd_cd.py (hunk 1 above)
Comment (jacobythwaites) [reply sparkl_cli/cmd_cd.py]: Well, it's arguable. I tried that, but I found myself using `sparkl cd` to show my current folder. I know in Linux you use `pwd` for that, i.e. a different command.

File: sparkl_cli/cmd_cd.py (hunk 2 above)
Comment (jacobythwaites) [reply sparkl_cli/cmd_cd.py]: As above

File: sparkl_cli/test/test_cd.py (hunk 3 above)
Comment (jacobythwaites) [reply sparkl_cli/test/test_cd.py]: Nice catch, thanks.

**Pull request merged**